addons/mail/__manifest__.py

```diff
@@ -2,7 +2,7 @@
 
 {
     'name': 'Discuss',
-    'version': '1.17',
+    'version': '1.18',
```

Comment:
Why do you upgrade the version when doing changes in master, and without a local upgrade script?

Reply:
Versioning is a convention ... we upgrade it when we change models, once / saas version. Anyway it has no impact as there is no clear usage of it ^^ .

Reply:
There is a clear usage of it: it runs the local module upgrade scripts when the module version changes

Do whatever you want with `mail`, but we use that in `account_*` so please don't bump the version when not needed.
It is just likely to bring inconsistencies with forward ports etc.

Reply:
Upgrade script is provided, is that was your question (maybe I did not understand it).

Reply:
You don't have anything in `addons/mail/migrations/1.17/`

Reply:
Is that a thing ? Seems only l10n have that kind of scripts. We provide our migration scripts in upgrade repo, as all standard upgrades. We don't do stable (local) upgrades, makes no sense for us.

Reply:
It is a thing if you want to run some scripts for a "light upgrade", meaning nothing that breaks the schema.
For instance, it could be some cleanup to be able to add a SQL constraint. (use case in website)
Or it could be creating some records for adding some company properties (use case in purchase)
Or like in accounting, to update the Chart of accounts of all companies when the law changes in various countries.

Reply:
Ok, did not know it existed to be honest (even upgrade never told me anything about that). Is it "standard" (like applied automatically when version changes) and/or automated ? Does it require -u, ... ?

Reply:
It works when doing `-u` or when clicking on `Upgrade` of the module in the UI.
Not sure if it works with `-i` but I expect it to.

Reply:
Ok. Really strange to have several ways of doing upgrades :shrug: It is strange that we change the version for minor upgrade and don't change it for major upgrades to be honest. Still thinking upgrades should be provided with module but I guess it is not going to change. Thanks for the info !

Reply:
IIRC it also allows custom modules to be upgraded on our platform (not sure)
And there a different ways because from a commercial PoV, upgrade is part of the OPW, so it needs to be on a private repo.
Note that the upgrade repo is (was) actually managed by just adding a symlink and using this mechanism :wink:

Reply:
Well that's an endless discussion, Antony is (was ?) globally in favor of releasing upgrade scripts, Fabien not (because money) ... typical never ending discussions allowing to make guru meetings even longer :D .

Anyway I could argue that upgrading the version number in master also makes sense, as when switching to the next version you will anyway migrate, just using upgrade code instead of local code. In the end don't have much strong opinion, if there is no crash I like to increase version number but if people starts yelling too much maybe I will change. Or not, because I like to make people yell. To think.